### ISSUE TITLE
updateEntityID function is not defined

### ISSUE THREAD

**Issue by sakairyota** [simplesamlphp/simplesamlphp#1703]
version: dev-master 80bb637
(PHP 8.1.1 / framework: laravel/framework 9.19)

When I try `requireAuth` in Simple class, I got the following error.
> Call to undefined method SimpleSAML\Metadata\MetaDataStorageHandlerPdo::updateEntityID()

This error occurred in MetaDataStorageHandlerPdo.php  L185.

`updateEntityID` is called in MetaDataStorageHandlerPdo, but updateEntityID function (originally it exists in MetaDataStorageSource) looks removed.

I also set `pdo` for metadata source. If the type is `flatfile`, it works. 
```
    'metadata.sources' => [
        ['type' => 'pdo'],
    ],
```
(one row exists in saml20_idp_remote table.)

code:
```
        $auth = new Simple('default-sp');
        $auth->requireAuth([
            'saml:idp' => 'https://localhost:4430/saml2/idp/metadata.php',
        ]);
```

stacktrace
```
SimpleSAML\Metadata\MetaDataStorageHandlerPdo: 185 getMetaData
SimpleSAML\Metadata\MetaDataStorageHandler: 317 getMetaData
SimpleSAML\ Metadata\MetaDataStorageHandler: 353 getMetaDataConfig
SimpleSAML\Modul \saml\Auth\ Source\SP : 305 getIdPMetadata
SimpleSAML\Module\saml\Auth\Source\SP: 663 startSSO
SimpleSAML\ Module\saml\Auth\Source\SP: 761 authenticate
SimpleSAML\Auth\Source: 195 initLogin
SimpleSAML\Auth\Simple: 167 login
SimpleSAML\Auth\Simple: 109 requireAuth
...
```

**Comment by tvdijen** [comment]
Interesting.. I guess you're the only one storing metadata in a database, because that method never existed..

**Comment by sakairyota** [comment]
I understand. updateEntityID is called in L132 and L185 of MetaDataStorageHandlerPdo, and only L132 is fixed in fc1ffee7388ae65289537ec1348d8b234068e50d.

According to the implementation of `updateEntityID ` deleted in fc1ffee7388ae65289537ec1348d8b234068e50d, following two codes will be same. 
```
$metadataSet[$d['entity_id']] = $this->updateEntityID($set, $entityId, $data);
```
and
```
$modifiedMetadataEntry = $entry; // copy value of $entry array to another array. (but not copy the reference of the array)
$modifiedMetadataEntry['entityID'] = $entityId;
$metadataSet[$d['entity_id']] = $modifiedMetadataEntry;
```

**Comment by tvdijen** [comment]
Thanks for the bug report!

**Comment by tvdijen** [comment]
Just to let you know; by mistake the fix didn't end up in yesterdays 2.0.0-rc2 release.
I made sure it will end up in the next (rc or final) release
